tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/AIRedactionalHelper.java

```diff
@@ -0,0 +1,76 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.tmail.mailet;
+
+import java.io.IOException;
+
+import jakarta.inject.Inject;
+
+import org.apache.mailet.MailetException;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import dev.ai4j.openai4j.OpenAiHttpException;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import reactor.core.publisher.Mono;
+
+
+public class AIRedactionalHelper {
+    private final AIBotConfig config;
+    private final ChatLanguageModel chatLanguageModel;
+    private final ChatLanguageModelFactory chatLanguageModelFactory;
+
+
+    @Inject
+    public AIRedactionalHelper(AIBotConfig config, ChatLanguageModelFactory chatLanguageModelFactory) {
+        this.config = config;
+        this.chatLanguageModelFactory = chatLanguageModelFactory;
+        this.chatLanguageModel = createChatLanguageModelModel(config);
+    }
+
+    public Mono<String> suggestContent(String userInput, String mailContent) throws OpenAiHttpException, MailetException, IOException {
```

Comment:
```suggestion
    public Mono<String> suggestContent(String userInput, Optional<String> mailContent) throws OpenAiHttpException, MailetException, IOException {
```

Does it have sens to call the redactional helper if I write a message from scratch?